nextcord/http.py

```diff
@@ -404,55 +1000,124 @@ async def close(self) -> None:
 
     # login management
 
-    async def static_login(self, token: str) -> user.User:
-        # Necessary to get aiohttp to stop complaining about session creation
-        self.__session = aiohttp.ClientSession(
-            connector=self.connector,
-            ws_response_class=DiscordClientWebSocketResponse,
-            headers={
-                "User-Agent": _USER_AGENT,
-            },
-        )
-        old_token = self.token
-        self.token = token
+    async def static_login(self, auth: str) -> user.User:
+        # TODO: Change this? This is literally just fetching /users/@me AKA "Get Current User", and is totally
+        #  usable with OAuth2. This doesn't actually have anything "log in" in any way.
+        self.set_default_auth(auth)
 
         try:
             data = await self.request(Route("GET", "/users/@me"))
         except HTTPException as exc:
-            self.token = old_token
             if exc.status == 401:
                 raise LoginFailure("Improper token has been passed.") from exc
             raise
 
         return data
 
-    def logout(self) -> Response[None]:
-        return self.request(Route("POST", "/auth/logout"))
+    async def exchange_access_code(
+        self,
+        *,
+        client_id: int,
+        client_secret: str,
+        code: str,
+        redirect_uri: str,
+        auth: Optional[str] = MISSING,
+        retry_request: bool = True,
+    ):
+        # This doesn't actually have hard ratelimits it seems? Not in the headers at least. The default bucket should
+        #  keep it at 1 every 1 second.
```

Comment:
This seems really slow. Maybe keep it at one at a time? Or hard-code oauth endpoints to not have RLs?

Reply:
I'm going to leave that for the OAuth PR.